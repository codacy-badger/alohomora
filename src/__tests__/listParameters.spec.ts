import listParametersPayload from '../__mocks__/describeParameters.json';
import SSM from '../__mocks__/aws-sdk/clients/ssm';
import { listParameters } from '../';

describe("listParameters", () => {
  it("request fails", async () => {
    const prefix = 'my-company/my-app';

    const handler = jest.fn()
      .mockImplementationOnce(() => { throw new Error('Some fatal error ocurred') });

    SSM.__setResponseForMethods({ describeParameters: handler });

    const response = await listParameters({ prefix });
    expect(response).toMatchSnapshot();

  });

  it("gets parameters", async () => {
    const prefix = 'my-company/my-app';

    const handler = jest.fn(() => ({ Parameters: listParametersPayload }));

    SSM.__setResponseForMethods({ describeParameters: handler });

    const response = await listParameters({ prefix });
    expect(response).toMatchSnapshot();
  });
  it("gets parameters, using nextToken", async () => {
    const prefix = 'my-company/my-app';

    const handler = jest.fn()
      .mockImplementationOnce(() => ({ Parameters: listParametersPayload.slice(0, 4), NextToken: 'MokeskinPouch' }))
      .mockImplementationOnce(() => ({ Parameters: listParametersPayload.slice(5, 9) }));

    SSM.__setResponseForMethods({ describeParameters: handler });

    const response = await listParameters({ prefix });
    expect(response).toMatchSnapshot();
    expect(handler).toHaveBeenCalledTimes(2);
  });

});
