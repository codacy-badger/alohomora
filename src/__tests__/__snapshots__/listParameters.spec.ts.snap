// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`listParameters gets parameters 1`] = `
"[90m┌───────────────────────────────────────────────[39m[90m┬─────────────[39m[90m┬────────────[39m[90m┬─────────────────────────────────────────────────┐[39m
[90m│[39m[31m Name                                          [39m[90m│[39m[31m Environment [39m[90m│[39m[31m Updated by [39m[90m│[39m[31m Updated at                                      [39m[90m│[39m
[90m├───────────────────────────────────────────────[39m[90m┼─────────────[39m[90m┼────────────[39m[90m┼─────────────────────────────────────────────────┤[39m
[90m│[39m GRAPHQL_ENDPOINT                              [90m│[39m all         [90m│[39m serious    [90m│[39m Thursday, November 29th, 2018, 1:26:47 PM (PST) [90m│[39m
[90m├───────────────────────────────────────────────[39m[90m┼─────────────[39m[90m┼────────────[39m[90m┼─────────────────────────────────────────────────┤[39m
[90m│[39m PAGERDUTY_E2E_STEP_FUNCTION_INTEGRATION_TOKEN [90m│[39m all         [90m│[39m Rubeus     [90m│[39m Monday, April 13th, 2020, 3:49:25 PM (PDT)      [90m│[39m
[90m├───────────────────────────────────────────────[39m[90m┼─────────────[39m[90m┼────────────[39m[90m┼─────────────────────────────────────────────────┤[39m
[90m│[39m PAGERDUTY_E2E_TEST_FAILURES_INTEGRATION_TOKEN [90m│[39m all         [90m│[39m Rubeus     [90m│[39m Tuesday, April 21st, 2020, 2:32:08 PM (PDT)     [90m│[39m
[90m├───────────────────────────────────────────────[39m[90m┼─────────────[39m[90m┼────────────[39m[90m┼─────────────────────────────────────────────────┤[39m
[90m│[39m PHABRICATOR_CONDUIT_TOKEN                     [90m│[39m all         [90m│[39m serious    [90m│[39m Friday, November 23rd, 2018, 11:25:09 AM (PST)  [90m│[39m
[90m├───────────────────────────────────────────────[39m[90m┼─────────────[39m[90m┼────────────[39m[90m┼─────────────────────────────────────────────────┤[39m
[90m│[39m PRISMA_ENDPOINT                               [90m│[39m all         [90m│[39m serious    [90m│[39m Friday, November 23rd, 2018, 11:26:02 AM (PST)  [90m│[39m
[90m├───────────────────────────────────────────────[39m[90m┼─────────────[39m[90m┼────────────[39m[90m┼─────────────────────────────────────────────────┤[39m
[90m│[39m PRISMA_MANAGEMENT_API_SECRET                  [90m│[39m all         [90m│[39m serious    [90m│[39m Tuesday, January 22nd, 2019, 12:46:29 PM (PST)  [90m│[39m
[90m├───────────────────────────────────────────────[39m[90m┼─────────────[39m[90m┼────────────[39m[90m┼─────────────────────────────────────────────────┤[39m
[90m│[39m SENTRY_DSN                                    [90m│[39m all         [90m│[39m serious    [90m│[39m Friday, November 23rd, 2018, 11:27:19 AM (PST)  [90m│[39m
[90m├───────────────────────────────────────────────[39m[90m┼─────────────[39m[90m┼────────────[39m[90m┼─────────────────────────────────────────────────┤[39m
[90m│[39m SENTRY_ENVIRONMENT                            [90m│[39m all         [90m│[39m serious    [90m│[39m Sunday, November 25th, 2018, 1:11:45 PM (PST)   [90m│[39m
[90m├───────────────────────────────────────────────[39m[90m┼─────────────[39m[90m┼────────────[39m[90m┼─────────────────────────────────────────────────┤[39m
[90m│[39m DEPLOY_SERVICE_STEP_FUNCTION_NAME             [90m│[39m development [90m│[39m severus    [90m│[39m Thursday, July 11th, 2019, 5:21:37 PM (PDT)     [90m│[39m
[90m├───────────────────────────────────────────────[39m[90m┼─────────────[39m[90m┼────────────[39m[90m┼─────────────────────────────────────────────────┤[39m
[90m│[39m DEPLOY_SERVICE_STEP_FUNCTION_WRAPPER_NAME     [90m│[39m development [90m│[39m severus    [90m│[39m Thursday, October 3rd, 2019, 12:44:53 PM (PDT)  [90m│[39m
[90m└───────────────────────────────────────────────[39m[90m┴─────────────[39m[90m┴────────────[39m[90m┴─────────────────────────────────────────────────┘[39m"
`;

exports[`listParameters gets parameters, using nextToken 1`] = `
"[90m┌───────────────────────────────────────────────[39m[90m┬─────────────[39m[90m┬────────────[39m[90m┬─────────────────────────────────────────────────┐[39m
[90m│[39m[31m Name                                          [39m[90m│[39m[31m Environment [39m[90m│[39m[31m Updated by [39m[90m│[39m[31m Updated at                                      [39m[90m│[39m
[90m├───────────────────────────────────────────────[39m[90m┼─────────────[39m[90m┼────────────[39m[90m┼─────────────────────────────────────────────────┤[39m
[90m│[39m GRAPHQL_ENDPOINT                              [90m│[39m all         [90m│[39m serious    [90m│[39m Thursday, November 29th, 2018, 1:26:47 PM (PST) [90m│[39m
[90m├───────────────────────────────────────────────[39m[90m┼─────────────[39m[90m┼────────────[39m[90m┼─────────────────────────────────────────────────┤[39m
[90m│[39m PAGERDUTY_E2E_STEP_FUNCTION_INTEGRATION_TOKEN [90m│[39m all         [90m│[39m Rubeus     [90m│[39m Monday, April 13th, 2020, 3:49:25 PM (PDT)      [90m│[39m
[90m├───────────────────────────────────────────────[39m[90m┼─────────────[39m[90m┼────────────[39m[90m┼─────────────────────────────────────────────────┤[39m
[90m│[39m PAGERDUTY_E2E_TEST_FAILURES_INTEGRATION_TOKEN [90m│[39m all         [90m│[39m Rubeus     [90m│[39m Tuesday, April 21st, 2020, 2:32:08 PM (PDT)     [90m│[39m
[90m├───────────────────────────────────────────────[39m[90m┼─────────────[39m[90m┼────────────[39m[90m┼─────────────────────────────────────────────────┤[39m
[90m│[39m PHABRICATOR_CONDUIT_TOKEN                     [90m│[39m all         [90m│[39m serious    [90m│[39m Friday, November 23rd, 2018, 11:25:09 AM (PST)  [90m│[39m
[90m├───────────────────────────────────────────────[39m[90m┼─────────────[39m[90m┼────────────[39m[90m┼─────────────────────────────────────────────────┤[39m
[90m│[39m PRISMA_MANAGEMENT_API_SECRET                  [90m│[39m all         [90m│[39m serious    [90m│[39m Tuesday, January 22nd, 2019, 12:46:29 PM (PST)  [90m│[39m
[90m├───────────────────────────────────────────────[39m[90m┼─────────────[39m[90m┼────────────[39m[90m┼─────────────────────────────────────────────────┤[39m
[90m│[39m SENTRY_DSN                                    [90m│[39m all         [90m│[39m serious    [90m│[39m Friday, November 23rd, 2018, 11:27:19 AM (PST)  [90m│[39m
[90m├───────────────────────────────────────────────[39m[90m┼─────────────[39m[90m┼────────────[39m[90m┼─────────────────────────────────────────────────┤[39m
[90m│[39m SENTRY_ENVIRONMENT                            [90m│[39m all         [90m│[39m serious    [90m│[39m Sunday, November 25th, 2018, 1:11:45 PM (PST)   [90m│[39m
[90m├───────────────────────────────────────────────[39m[90m┼─────────────[39m[90m┼────────────[39m[90m┼─────────────────────────────────────────────────┤[39m
[90m│[39m DEPLOY_SERVICE_STEP_FUNCTION_NAME             [90m│[39m development [90m│[39m severus    [90m│[39m Thursday, July 11th, 2019, 5:21:37 PM (PDT)     [90m│[39m
[90m└───────────────────────────────────────────────[39m[90m┴─────────────[39m[90m┴────────────[39m[90m┴─────────────────────────────────────────────────┘[39m"
`;

exports[`listParameters request fails 1`] = `""`;
